keps/sig-scheduling/5710-workload-aware-preemption/README.md

```diff
@@ -0,0 +1,1064 @@
+   when preemption is triggerred but the pod/workload that triggerred it in the end cannot
+   be bound anyway due to inability to schedule the workload as a whole.
+
+The rest of this KEP explains those pieces in more detail.
+
+### User Stories
+
+#### Preemption of AI Training job
+
+When running an AI Training job, I want to ensure that it will not be partially preempted.
+If at least one my pods is not running, the others are not making progress anyway and are
+just wasting the resources in the cluster.
+
+#### Preemption of Multihost Inference
+
+When running multihost inference using LeaderWorkerSet, I want to ensure that its single
+replica (one leader and N workers) will not be partially preempted. If at least one of
+such pods is not running, the others are not able to serve and are just wasting resources
+in the cluster.
+
+#### Preemption of Multihost Inference that can run in a degraded mode
+
+When running multihost inference using LeaderWorkerSet that can run in a degraded mode,
+I don't want to preempt the whole replica (one leader and N workers) if a single worker
+would be preempted, because I prefer to serve in a degraded mode than be completely
+disrupted.
+
+#### Preemption cost
+
+When a long-running AI Training job, its cost of preemption differs over time and depends
+on how long ago it checkpointed its state. I want to be able to somehow influence the
+preemption priority of my job based on how much would it really cost me.
+
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+### Preemption unit
+
+We start with defining what is a unit of preemption. While we can imagine usecases with preemption
+unit being an arbitrary group of pods, in majority of real world usecases this is actually aligned
+with the scheduling unit. In other words, the group of pods that should be preempted together matches
+a group that was initially scheduled together as a gang.
+
+In the future, we may want to support usecases when a single scheduling unit consists of multiple
+preemption groups, but we leave that usecase as a future extension (it can be addressed when we
+decide to extend Workload API with PodSubGroup concept - for more details see
+[API Design For Gang and Workload-Aware Scheduling]). However, we never expect preemption unit to
+be larger than scheduling unit.
```

Comment:
Perhaps define "Workload Portion" as one of {all pods in Workload, all pods in a PodGroup replica, a single Pod}.
Then define schedulingUnit and preemptionUnit as being Workload Portions.
This makes it clear that, for this KEP, we don't intend to support arbitrary lists of pods as scheduling or preemption units